File: backend/app/schemas/financial_summary.py
"""Pydantic schemas for the OZET financial summary."""
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class FinancialSummaryRead(BaseModel):
    id: int
    project_id: int
    company_label: str
    as_of_date: date

    isveren_tahsilatlari: Decimal
    firma_odemeleri: Decimal
    ucret_giderleri: Decimal
    vergi_odemeleri: Decimal
    gelir_vergisi: Decimal
    kdv: Decimal
    faiz_gelirleri: Decimal
    banka_giderleri: Decimal
    diger_gelir_giderler: Decimal
    toplam: Decimal

    source_filename: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/financial-summary.ts
export interface FinancialSummary {
  id: number;
  project_id: number;
  company_label: string; // "Monotek" | "Monart"
  as_of_date: string; // ISO yyyy-mm-dd

  isveren_tahsilatlari: string; // decimals serialized as strings
  firma_odemeleri: string;
  ucret_giderleri: string;
  vergi_odemeleri: string;
  gelir_vergisi: string;
  kdv: string;
  faiz_gelirleri: string;
  banka_giderleri: string;
  diger_gelir_giderler: string;
  toplam: string;

  source_filename: string | null;
  created_at: string;
  updated_at: string;
}
